### ISSUE TITLE
onnx.shape_inference.infer_shapes   exit

### ISSUE THREAD

**Issue by liulai** [onnx/onnx#2976]
# Bug Report
onnx.shape_inference.infer_shapes is invalid. The program exit automatically.

### System information

- OS Platform and Distribution: Windows 10

- ONNX version: 1.7.0

- Python version: 3.7.2

### Describe the bug

> ```
>F:\DATASET\MPI-Sintel-complete\onnx>ipython
>Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)]
>Type 'copyright', 'credits' or 'license' for more information
>IPython 7.4.0 -- An enhanced Interactive Python. Type '?' for help.
>
>In [1]: import onnx
>
>In [2]: PATH = r'F:\DATASET\MPI-Sintel-complete'
>
>In [3]: model = onnx.load(PATH+'/onnx/reconet.onnx')
>
>In [4]: inferred_model = onnx.shape_inference.infer_shapes(model)
>
>F:\DATASET\MPI-Sintel-complete\onnx>
> ```

![image](https://user-images.githubusercontent.com/12567867/91354383-4192ee80-e81f-11ea-8257-7a944f363998.png)

model file:[reconet.zip](https://github.com/onnx/onnx/files/5132703/reconet.zip)

**Comment by jcwchen** [comment]
Hi @liulai,
Similar to https://github.com/onnx/onnx/issues/2965.
Sorry that current shape inference does not catch the exception properly. 
This segmentation fault bug will be fixed in this PR: https://github.com/onnx/onnx/pull/2783.
You can get the latest onnx from master and use `onnx.shape_inference.infer_shapes` again.

I got the error as follows:
```
Shape inference error(s): (op_type:Sub, node name: Sub_4): [TypeInferenceError] Element type of input 0 unknown
(op_type:Sub, node name: Sub_26): [TypeInferenceError] Element type of input 0 unknown
(op_type:Sub, node name: Sub_48): [TypeInferenceError] Element type of input 0 unknown
(op_type:Sub, node name: Sub_129): [TypeInferenceError] Element type of input 0 unknown
(op_type:Sub, node name: Sub_180): [TypeInferenceError] Element type of input 0 unknown
(op_type:Sub, node name: Sub_202): [TypeInferenceError] Element type of input 0 unknown
```

**Comment by liulai** [comment]
thank you very much, I solved the problem, the inference is not tensor of graph, but tensor_value_info. according to tensors, I create corresponding tensor_value_info.

**Comment by liulai** [comment]
> Hi @liulai,
> Similar to #2965.
> Sorry that current shape inference does not catch the exception properly.
> This segmentation fault bug will be fixed in this PR: #2783.
> You can get the latest onnx from master and use `onnx.shape_inference.infer_shapes` again.
> 
> I got the error as follows:
> 
> ```
> Shape inference error(s): (op_type:Sub, node name: Sub_4): [TypeInferenceError] Element type of input 0 unknown
> (op_type:Sub, node name: Sub_26): [TypeInferenceError] Element type of input 0 unknown
> (op_type:Sub, node name: Sub_48): [TypeInferenceError] Element type of input 0 unknown
> (op_type:Sub, node name: Sub_129): [TypeInferenceError] Element type of input 0 unknown
> (op_type:Sub, node name: Sub_180): [TypeInferenceError] Element type of input 0 unknown
> (op_type:Sub, node name: Sub_202): [TypeInferenceError] Element type of input 0 unknown
> ```

thank you very much, I solved the problem, the inference is not tensor of graph, but tensor_value_info. according to tensors, I create corresponding tensor_value_info.
